src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+    }
+    // Disable ocarina
+    func_800ED858(0); // Comment from z_en_fu.c: seems to be related to setting instrument type
+    globalCtx->msgCtx.msgMode = 55;
+    this->actionFunc = EnFr_PrintTextBox;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B464.s")
+void EnFr_PrintTextBox(EnFr* this, GlobalContext* globalCtx) {
+    // Looking up text box message
+    func_8010B680(globalCtx, this->actor.textId, &this->actor);
+    this->actionFunc = EnFr_TalkBeforeReward;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B524.s")
+void EnFr_TalkBeforeReward(EnFr* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && func_80106BC8(globalCtx)) {
+        this->frogSongTimer = 100;
+        func_80106CCC(globalCtx);
+        this->actionFunc = EnFr_SetReward;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B638.s")
+void EnFr_SetReward(EnFr* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B778.s")
+    u16 songIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B8D0.s")
+    sEnFrPointers.flags = 0xC;
+    songIndex = this->songIndex;
+    this->actionFunc = EnFr_Deactivate;
+    this->reward = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1B9D8.s")
+    // 5 Child Song Rewards: 1st time Purple Rupee, afterwards blupee
+    if ((songIndex >= 0) && (songIndex < 5)) {
+        if (!(gSaveContext.eventChkInf[13] & sSongIndex[songIndex])) {
+            gSaveContext.eventChkInf[13] |= sSongIndex[songIndex];
+            this->reward = GI_RUPEE_PURPLE;
+        } else {
+            this->reward = GI_RUPEE_BLUE;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BB34.s")
+    // Song of Storms Reward: 1st time HP, afterwards blupee
+    else if (songIndex == 5) {
+        if (!(gSaveContext.eventChkInf[13] & sSongIndex[songIndex])) {
+            gSaveContext.eventChkInf[13] |= sSongIndex[songIndex];
+            this->reward = GI_HEART_PIECE;
+        } else {
+            this->reward = GI_RUPEE_BLUE;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BC14.s")
+    // Frogs 2 Reward: 1st time HP, afterwards purple rupee
+    else if (songIndex == 6) {
+        if (!(gSaveContext.eventChkInf[13] & sSongIndex[songIndex])) {
+            gSaveContext.eventChkInf[13] |= sSongIndex[songIndex];
+            this->reward = GI_HEART_PIECE;
+        } else {
+            this->reward = GI_RUPEE_PURPLE;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BCB4.s")
+void EnFr_Deactivate(EnFr* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1BD94.s")
+    EnFr* frog_loop1;
+    EnFr* frog_loop2;
```

Comment:
do these need separate temps?

Reply:
It doesn't match if there isn't two temps. a1, v0, v1 all get swapped if only 1 temp is used. Although there might be a solution with 1 temp, I just didn't find it.